src/chainstate/coordinator/tests.rs

```diff
@@ -2852,9 +3231,539 @@ fn test_epoch_switch_cost_contract_instantiation() {
+                .with_read_only_clarity_tx(
+                    &sort_db.index_conn(),
+                    &StacksBlockId::new(&stacks_tip.0, &stacks_tip.1),
+                    |conn| {
+                        conn.with_clarity_db_readonly(|db| {
+                            db.get_stacks_epoch(burn_block_height as u32).unwrap()
+                        })
+                    },
+                )
+                .unwrap()
+                .block_limit
+                .runtime,
+            expected_runtime
+        );
+
+        // check that pox-2 contract DNE before epoch 2.1, and that it does exist after
+        let does_pox_2_contract_exist = chainstate
+            .with_read_only_clarity_tx(
+                &sort_db.index_conn(),
+                &StacksBlockId::new(&stacks_tip.0, &stacks_tip.1),
+                |conn| {
+                    conn.with_clarity_db_readonly(|db| {
+                        db.get_contract(&boot_code_id(POX_2_NAME, false))
+                    })
+                },
+            )
+            .unwrap();
+
+        if burn_block_height < 8 {
+            assert!(does_pox_2_contract_exist.is_err())
+        } else {
+            assert!(does_pox_2_contract_exist.is_ok())
+        }
+    }
+}
+
+fn get_total_stacked_info(
+    chainstate: &mut StacksChainState,
+    burn_dbconn: &dyn BurnStateDB,
+    parent_tip: &StacksBlockId,
+    reward_cycle: u64,
+    is_pox_2: bool,
+) -> Result<u128, InterpreterError> {
+    chainstate
+        .with_read_only_clarity_tx(burn_dbconn, parent_tip, |conn| {
+            conn.with_readonly_clarity_env(
+                false,
+                CHAIN_ID_TESTNET,
+                ClarityVersion::Clarity2,
+                PrincipalData::parse("SP3Q4A5WWZ80REGBN0ZXNE540ECJ9JZ4A765Q5K2Q").unwrap(),
+                None,
+                LimitedCostTracker::new_free(),
+                |env| {
+                    let eval_str = format!(
+                        "(contract-call? '{}.{} get-total-ustx-stacked u{})",
+                        &boot_code_addr(false),
+                        if is_pox_2 { POX_2_NAME } else { POX_1_NAME },
+                        reward_cycle
+                    );
+
+                    let result = env.eval_raw(&eval_str).map(|v| v.expect_u128());
+                    Ok(result)
+                },
+            )
+            .unwrap()
+        })
+        .unwrap()
+}
+
+// This test verifies that the correct contract is used for PoX for stacking operations.
+// Need to ensure that after v1_unlock_height, stacking operations are executed in the "pox-2" contract.
+// After the transition to Epoch 2.1 but before v1_unlock_height, stacking operations that are
+// sent should occur in the "pox.clar" contract.
+#[test]
+fn test_epoch_verify_active_pox_contract() {
+    let path = "/tmp/stacks-blockchain-epoch-switch-pox-contract-instantiation";
+    let _r = std::fs::remove_dir_all(path);
+
+    let pox_v1_unlock_ht = 12;
```

Comment:
This is the same height as the epoch transition.  Is it the case on mainnet that the v1 unlock height and the epoch transition will be the same?

Reply:
Hmm, it's the same height as the first 2.1 reward cycle I believe, but the actual epoch switch occurs at burn block height 8 in this test.